backend/src/test/java/gov/cdc/usds/simplereport/api/BaseFullStackTest.java

```diff
@@ -35,6 +36,7 @@
   @Autowired private DbTruncator _truncator;
   @Autowired private AuditService _auditService;
   @Autowired protected TestDataFactory _dataFactory;
+  @MockBean private CurrentOrganizationRolesContextHolder _orgRoleContextHolder;
```

Comment:
This ... seems like a problem? I assume this is needed because the setup code calls getCurrentOrganizationRoles and so we're not in a request context and everything goes to hell? But that means that we're not testing that behavior at all.

Reply:
You're correct, that's exactly the reason for this! I started to mock up a way to explicitly test the context holder, but it seems that in order to test request-scoped beans, you have to spin up the entire web app context. We've avoided this with most of our tests to date because it's expensive, and the java tests already take several minutes to run.

What do you recommend?

Reply:
Couldn't you just drop the context holder? It's caching some light processing of the already-cached Spring security context.

If some amount of caching is required, it might be more straightforward to just use a `WeakMap<Authentication, List<OrganizationRoleClaims>>` as a cache in https://github.com/CDCgov/prime-simplereport/blob/main/backend/src/main/java/gov/cdc/usds/simplereport/service/LoggedInAuthorizationService.java#L38

Reply:
The processing includes a couple database calls, I think.

Reply:
That's right – the context holder exists to eliminate a number of duplicate queries to the Organization and Facility repositories that were contributing to the N+1 explosion in `GetPatientsCountByFacility`. And the trouble here is that in the test _setup_ code, there isn't a request context in which to cache these results in the first place.

Reply:
Addressed in a946c41d